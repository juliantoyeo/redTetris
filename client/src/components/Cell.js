import React from 'react';
import PropTypes from 'prop-types';
import { PIECES } from '../constants/gameConstant';

const mainContainerStyle = (props) => {
	let borderWidth = 0.4;
	if (props.numberOfPlayer > 3)
		borderWidth = 0.3;
	return ({
		width: 'auto',
		background: `rgba(${props.color}, 0.8)`,
		border: `${props.type === '0' ? '1px solid' : `${borderWidth}vw solid`}`,
		borderColor: `rgba(${props.color}, 1)`
		// borderBottomColor: `rgba(${props.color}, 0.1)`,
		// borderRightColor: `rgba(${props.color}, 1)`,
		// borderTopColor: `rgba(${props.color}, 1)`,
		// borderLeftColor: `rgba(${props.color}, 0.3)`
	});
}

const Cell = (props) => {
	const { type, numberOfPlayer } = props;
	return (
		<div style={mainContainerStyle({ type, color: PIECES[type].color, numberOfPlayer: numberOfPlayer })} className={'cell'}/>
	)
}

Cell.propTypes = {
	type: PropTypes.string,
	numberOfPlayer: PropTypes.number
};

export default React.memo(Cell);