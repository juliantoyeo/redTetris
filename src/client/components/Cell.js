import React from 'react'
import { PIECES } from '../constants/gameConstant'

const mainContainerStyle = (props) => {
	return ({
		width: 'auto',
		background: `rgba(${props.color}, 0.8)`,
		border: `${props.type === '0' ? '1px solid' : '0.6vw solid'}`,
		borderColor: `rgba(${props.color}, 1)`,
		// borderBottomColor: `rgba(${props.color}, 0.1)`,
		// borderRightColor: `rgba(${props.color}, 1)`,
		// borderTopColor: `rgba(${props.color}, 1)`,
		// borderLeftColor: `rgba(${props.color}, 0.3)`
	})
}

const Cell = (props) => {
	const { type } = props
	return (
		<div style={mainContainerStyle({ type, color: PIECES[type].color })} className={"cell"}/>
	)
}

export default React.memo(Cell)