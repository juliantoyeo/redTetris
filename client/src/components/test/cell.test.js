import React from 'react';
import { shallow } from 'enzyme';

import Cell from '../Cell';

describe('Test Cell component', () => {
	it('Should render properly', () => {
		const wrapper = shallow(<Cell type={'0'} numberOfPlayer={1} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render properly for 4 player', () => {
		const wrapper = shallow(<Cell type={'I'} numberOfPlayer={4} />);
		expect(wrapper).toMatchSnapshot();
	})
});