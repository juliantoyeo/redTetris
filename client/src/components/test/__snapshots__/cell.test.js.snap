// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cell component Should render properly 1`] = `
<div
  className="cell"
  style={
    Object {
      "background": "rgba(0, 0, 0, 0.8)",
      "border": "1px solid",
      "borderColor": "rgba(0, 0, 0, 1)",
      "width": "auto",
    }
  }
/>
`;

exports[`Test Cell component Should render properly for 4 player 1`] = `
<div
  className="cell"
  style={
    Object {
      "background": "rgba(186,225,255, 0.8)",
      "border": "0.3vw solid",
      "borderColor": "rgba(186,225,255, 1)",
      "width": "auto",
    }
  }
/>
`;
